apps/sms/views/shared/js/settings.js

```diff
@@ -135,6 +135,26 @@ var Settings = {
     }.bind(this));
   },
 
+  // return true if send-read-report switch is ON else false
+  toSendReadReport: function toSendReadReport() {
+    if (!navigator.mozSettings) {
+      return Promise.reject(new Error('The mozSettings API is not available.'));
+    }
+
+    var defer = Utils.Promise.defer();
+    var key = 'message.mms.sendReadReport.enabled';
+    var request = navigator.mozSettings.createLock().get(key);
+    request.onsuccess = function onsuccess(e) {
+      defer.resolve(e.target.result);
+    };
+
+    request.onerror = function onerror(e) {
+      defer.reject(e.target.error);
+    };
+
+    return defer.promise;
+  },
```

Comment:
in https://bugzilla.mozilla.org/show_bug.cgi?id=1180470#c19 I proposed to synchronize the setting value using similar methods that we have already (for example look `initSmsMaxConcatenatedMsg` and `keyHandlerSet` in `init`).

Then I thought that we can have a property "Settings.sendReadReport" that is a promise that resolves to the value. This way it's not resolved at startup yet, but the code in MessageManager can still request the promise and use the value when it's resolved.

So in MessageManager you can have:
`Settings.sendReadReport.then((value) => ...)`